### PR TITLE
Course Change - Add the study mode wizard

### PR DESCRIPTION
**Pull request by JR-G**
## Context

Currently providers can change the details of a course at the point of offer. We'd like to let providers do this before the point of offer.

Following from [#6585](https://github.com/DFE-Digital/apply-for-teacher-training/pull/6585) this is the third PR in the series which will allow the user to change the study mode.

This stage of the flow will be skipped if the user picks an option which doesn't have both modes available to select

## Changes proposed in this pull request

image

## Guidance to review

- Should not be preselected if they've changed course

## Link to Trello card

https://trello.com/c/E97RWJnY

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/forms/provider_interface/course_wizard.rb [hunk 1]
```diff
@@ -84,5 +94,12 @@ def save_and_go_to_next_step(step)
 
       next_step(step)
     end
+
+    def sanitize_attrs(attrs)
+      if !last_saved_state.empty? && attrs[:course_id].present? && last_saved_state['course_id'] != attrs[:course_id]
+        attrs.merge!(study_mode: nil, course_option_id: nil)
+      end
+      attrs
+    end
```
Comment (SamJamCul) [inline app/forms/provider_interface/course_wizard.rb:104]: What's happening in this step?

File: spec/system/provider_interface/provider_changes_a_course_before_point_of_offer_spec.rb [hunk 2]
```diff
@@ -18,6 +18,74 @@
   end
   let(:course_option) { build(:course_option, course: course) }
 
+  # TODO: Uncomment these once location is built
+
+  # scenario 'Changing a course choice before point of offer (no study mode or location)' do
```
Comment (cwrw) [inline spec/system/provider_interface/provider_changes_a_course_before_point_of_offer_spec.rb:23]: should we instead only test the full happy path in system spec and the edge-cases in unit tests?

File: app/controllers/provider_interface/courses/courses_controller.rb [hunk 3]
```diff
@@ -14,7 +14,11 @@ def update
         if @wizard.valid_for_current_step?
           @wizard.save_state!
 
-          redirect_to [:edit, :provider_interface, @application_choice, :offer, @wizard.next_step]
+          if @wizard.next_step.nil?
+            redirect_to provider_interface_applications_path(@application_choice)
```
Comment (cwrw) [inline app/controllers/provider_interface/courses/courses_controller.rb:18]:
```suggestion
            redirect_to provider_interface_application_choice_path(@application_choice)
```

Comment (JR-G) [reply app/controllers/provider_interface/courses/courses_controller.rb:18]: 👌

**Review by cwrw — APPROVED** [review]
